### ISSUE TITLE
Find Usages is not working with cmd + click when plugin is active

### ISSUE THREAD

**Issue by pySilver** [InSyncWithFoo/pyright-for-pycharm#72]
### What happened?

This `Find Usages` popup is not triggered with `cmd + click` when plugin is active.

![CleanShot 2024-09-19 at 00 05 34@2x](https://github.com/user-attachments/assets/2750c752-47b7-4628-9805-ee44364c6cd7)

### Steps to reproduce

Activate plugin & Try to navigate with cmd + click.

### Operating system

macOS

**Comment by angelozerr** [comment]
> This is possibly a limitation rather than a bug (could you confirm, @angelozerr?).

I confirm that. As you said Ctrl+Click consumes `textDocument/definition` which is a commons usecase for any language server.

It could be doable to consumes other LSP request like `textDocument/typeDefinition`, `textDocument/declaration`, `textDocument/reference` but according language server some LSP requets could be slow (ex : `textDocument/reference`). It is an another LSP API usecase to customize that,please create an issue in LSP4IJ.

**Comment by pySilver** [comment]
@InSyncWithFoo yep, exactly I'm using LSP4IJ mode. It's just a lifelong habit to use this shortcut for me. So yeah, kinda hard to switch. I'm also a little bit confused with your second plugin which get updates but marked as "will be merged with `pyright-for-pycharm` soon". Which one should I use giving that I use paid PyCharm? The other one uses language server and seems to support cmd+click fine.

**Comment by angelozerr** [comment]
> The other one uses language server and seems to support cmd+click fine.

You mean that with the LSP support from Jetbrains, cmd+click open references? @InSyncWithFoo can you confirm that please (I cannot test since I have no license for Ultimate)

If it is working, @InSyncWithFoo I think LSP4IJ should provide to customize this cmd+click (by defualt just definition), but you could customize it to have reference too,please create an issue in LSP4IJ.

**Comment by pySilver** [comment]
@angelozerr I believe it is lsp support from jetbrains as it does not require any additional plugins (LSP4IJ)

**Comment by angelozerr** [comment]
LSP support from Jetbrains is not free and it is not available for IJ Community, it is one reason why we have created LSP4IJ

**Comment by pySilver** [comment]
@angelozerr Understood.

@InSyncWithFoo Just wanted to confirm that this plugin does not trigger usage window in either console or LSP4IJ modes. (Unlike it's sister plugin). Both shortcuts (Find Usages Popup: `cmd + opt + f7`) and (Go to LSP References `opt + cmd + H`) works but aren't triggered with `cmd + click`.

**Comment by pySilver** [comment]
@InSyncWithFoo hm. I've tested once again and yes I can confirm that in command line mode `find usages` panel works

**Comment by yang** [comment]
I'm a little confused about the take-away from this thread. Some basic questions:

- I am using Pycharm paid, so that means I have access to the (paid) native LSP feature as well as the community LSP4IJ?
- Does one of these generally work "better"?
- Does one of these support the Usages functionality being brought up in this thread?
- How do I get the Pyright plugin to use one vs. the other? I see that I have a LSP4IJ plugin installed for some reason (I may have installed it a long time ago?), do I need to uninstall that to try out native LSP?

Thanks.

**Comment by yang** [comment]
I don't seem to see a way to choose Native, there's only LSP4IJ listed here:

<img width="1065" alt="Image" src="https://github.com/user-attachments/assets/c139fd88-43ce-4835-8124-5b2a9da9bc6d" />

When I try to disable LSP4IJ:

<img width="1526" alt="Image" src="https://github.com/user-attachments/assets/204a6d79-ab09-4600-84f4-9f0acd631cb8" />

**Comment by yang** [comment]
Ah, thanks - I had upgraded but it was only taking smaller steps within 2024.1 first. Had to check for upgrades a couple more times.
